src/Workspaces/Core/Portable/Workspace/ProjectSystem/ProjectSystemProject.cs

```diff
@@ -970,10 +982,18 @@ public void AddAnalyzerReference(string fullPath)
 
         var mappedPaths = GetMappedAnalyzerPaths(fullPath);
 
+        bool containsSdkCodeStyleAnalyzers;
+
         using var _ = CreateBatchScope();
 
         using (_gate.DisposableWait())
         {
+            // Track the unmapped analyzer paths
+            _unmappedProjectAnalyzerPaths.Add(fullPath);
+
+            // Determine if we started using SDK CodeStyle analyzers while access to _unmappedProjectAnalyzerPaths is gated.
+            containsSdkCodeStyleAnalyzers = _unmappedProjectAnalyzerPaths.Any(IsSdkCodeStyleAnalyzer);
```

Comment:
Can we do `containsSdkCodeStyleAnalyzers = _hasSdkCodeStyleAnalyzers | IsSdkCodeStyleAnalyzer(fullPath)` instead of enumerating all the paths each time?

Reply:
agree on tweaking this.

Reply:
Changing this to only track the codestyle analyzer paths instead of tracking all analyzer paths. Only one call to `IsSdkCodeStyleAnalyzer` when adding or removing.